acra/guides/advanced-integrations/client-side-integration-with-acra-connector.md

```diff
@@ -0,0 +1,224 @@
+```
+
+The name of the key should be longer than 5 characters. It is also used as an identifier for the [Secure Session]({{<ref "themis/crypto-theory/cryptosystems/secure-session.md" >}}) connection between AcraConnector and AcraServer.
+
+The generated keypair `client_name` and `client_name.pub` will appear in .acrakeys (or anywhere you ask with `--keys_output_dir=/path/to/dir` argument).
+
+- Exchange public keys:
+
+    - You should put public key `client_name.pub` into the corresponding folder (.acrakeys) on AcraServer.
+    - You should put AcraServer's public key (called `client_name_server.pub`) to AcraConnector's key folder (`.acrakeys` or anything you chose in `--keys_output_dir`).
+
+Now, proceed to [launching AcraConnector]({{< ref "acra/configuring-maintaining/general-configuration/acra-connector.md#launching-acraconnector-INVALID" >}}).
+
+⚙️**Not sure how to configure AcraConnector? Check out the [engineering examples](https://github.com/cossacklabs/acra-engineering-demo/)!** ⚙️
+
+### Launching AcraConnector
+
+By default, AcraConnector is ready to talk to AcraServer. You need to use a one-line command:
+
+```
+$GOPATH/bin/acra-connector --client_id=client_name --acraserver_connection_host=acra.server.host
+```
+
+To point AcraConnector to AcraTranslator, configure an appropriate connection host/port/string and mode:
+
+```
+$GOPATH/bin/acra-connector --client_id=client_name --acratranslator_connection_host=acra.translator.host --mode=acratranslator
+```
+
+For security reasons, consider configuring your firewall to allow the connections only from legit AcraConnector IPs. If an attacker compromises your client application and AcraConnector, filtering IP addresses might prevent DoS.
+
+You can get more details about AcraConnector configuration [here]({{< ref "acra/configuring-maintaining/general-configuration/acra-connector.md" >}}).
+
+
+### AcraWriter
+
+After you have configured AcraConnector, your application can keep using your database handling code as before — all the extra work will now be taken care of by Acra's components.
+
+AcraWriter is a library for your code that can be used anywhere within your app whenever you need to encrypt sensitive records.
+
+Under the hood, it is basically the [Themis]({{< ref "themis/" >}}) library generating AcraStructs with the keys you've provided to AcraWriter.
+
+To start protecting your data, pick a place in your code where and integrate AcraWriter library (we support 8 languages, see the [Building AcraWriter for your language]({{< ref "acra/configuring-maintaining/installing/building-acrawriter.md" >}}) section). Don't see your language? Write your own [AcraStruct]({{< ref "acra/acra-in-depth/data-structures/#understanding-acrastruct" >}}) encrypter, it's easy!
+
+
+If you plan to use Acra with:
+
+- PostgreSQL — use BYTEA binary type.
+
+- MySQL — use mysql binary type.
+
+
+### Acra example usage scenario
```

Comment:
This subsection is actually our philosophy. If we want it to keep, may be better place for it is somewhere at the beginning of this document (after explaining AcraConnector, AcraWriter terms), not among direct engineering instructions and commands, isn't it?

Reply:
At the beginning we explain why user needs AcraConnector, what is it, and why we provide it. Here we just remind how it works in general before examples. In context with AcraConnector and additional note that encryption doesn't require AcraConnector, only AcraWriter. I'm not sure that it's inappropriate place for that